flake.nix

```diff
@@ -74,17 +66,29 @@
       name = "tailscale";
 
       src = ./.;
-      vendorSha256 = fileContents ./go.mod.sri;
-      nativeBuildInputs = pkgs.lib.optionals pkgs.stdenv.isLinux [ pkgs.makeWrapper pkgs.git ];
```

Comment:
I don't know why `pkgs.git` was in here, so I'm not confident something won't break if we remove it.

Reply:
`nativeBuildInputs` should only affect the build of the package. I was able to build it on x86_64-linux and run the systemd service both in a VM and on my system just fine. I'll be able to test on aarch64-darwin (the build at least) shortly, if that would reduce your hesitation.

That said I'll happily revert that commit if you'd rather not deal with that!

Reply:
(note it's possible I screwed up the version embedding when I wrote this flake and it doesn't work correctly even with pkgs.git. If that is the case, pls remove the build input and I'll file a bug to figure out how to embed the right version info)

Reply:
Both with and without report
```
❯ ./result/bin/tailscale version
1.51.0-ERR-BuildInfo
  track: unstable (dev); frequent updates and bugs are likely
  go version: go1.21.0
```

Reply:
Yeah, I don't think the version embedding works anymore:

```
# tip of this branch
vin@scadrial ~/w/v/tailscale (cole/ds-1301-fix-tailscale-flake) 0.6s> ./result/bin/tailscale --version
1.51.0-ERR-BuildInfo
  track: unstable (dev); frequent updates and bugs are likely
  go version: go1.21.0

# tip of main, with the go.mod.sri fixed up in a new commit so it will actually build
vin@scadrial ~/w/v/tailscale (main) 10.8s> result/bin/tailscale --version
1.51.0-ERR-BuildInfo
  track: unstable (dev); frequent updates and bugs are likely
  go version: go1.21.0
```

Reply:
If it's not working on main then it's fine to file a follow-up issue to address flake version embedding.

Reply:
Yup, I can file a bug for that, in the meantime removing pkgs.git is the right thing. Thanks!

Reply:
@danderson I think I figured it out: it looks like the new way to set the git hash is through `version.gitCommitStamp`:

```
# with another change that makes the dirtyRev show up
vin@scadrial ~/w/v/tailscale (main) 10.3s> result/bin/tailscale --version
1.51.0-ERR-BuildInfo
  track: unstable (dev); frequent updates and bugs are likely
  tailscale commit: 087260734b5d513452cbb8baefef86c8e5bcd3ee-dirty
  go version: go1.21.0
```

Would you like me to file an issue and then PR the fix?

```diff
diff --git a/flake.nix b/flake.nix
index 9a496db7..d60069fc 100644
--- a/flake.nix
+++ b/flake.nix
@@ -44,7 +44,7 @@
     # tailscaleRev is the git commit at which this flake was imported,
     # or the empty string when building from a local checkout of the
     # tailscale repo.
-    tailscaleRev = if builtins.hasAttr "rev" self then self.rev else "";
+    tailscaleRev = self.rev or self.dirtyRev or "dirty";
     # tailscale takes a nixpkgs package set, and builds Tailscale from
     # the same commit as this flake. IOW, it provides "tailscale built
     # from HEAD", where HEAD is "whatever commit you imported the
@@ -68,7 +68,7 @@
       src = ./.;
       vendorSha256 = pkgs.lib.fileContents ./go.mod.sri;
       nativeBuildInputs = pkgs.lib.optionals pkgs.stdenv.isLinux [ pkgs.makeWrapper ];
-      ldflags = ["-X tailscale.com/version.GitCommit=${tailscaleRev}"];
+      ldflags = ["-X tailscale.com/version.gitCommitStamp=${tailscaleRev}"];
       CGO_ENABLED = 0;
       subPackages = [ "cmd/tailscale" "cmd/tailscaled" ];
       doCheck = false;

```

Reply:
Ah yeah, I moved the version embedding stuff around at some point and forgot to fix the flake :facepalm: A PR would be great if you have the time/energy! One note, instead of setting the commit stamp to "dirty", you could instead leave the stamp empty and set `tailscale.com/version.gitDirtyStamp=true`, that way the version presentation logic will understand what's happening instead of trying to treat a literal `dirty` as a git stamp.

Reply:
Understood! I think I'll have to leave out the `gitDirtyStamp` (and `self.dirtyRev`) change, since the variable is a boolean and Go doesn't appear to allow setting that in ldflags:

```
tailscale.com/version.gitDirtyStamp: cannot set with -X: not a var of type string (type:bool)
```

(Even if I set it to the "strings" `1` or `true`, it still complains)

Reply:
Done in https://github.com/tailscale/tailscale/pull/9997!